File: benchmarks/jasboot_vs_v8/collections_large.c
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>

int main(void) {
    int *valores = (int *)malloc(sizeof(int) * 20000);
    int *indices = (int *)malloc(sizeof(int) * 20000);
    long long acc = 0;
    if (!valores || !indices) return 1;

    for (int i = 0; i < 20000; i++) {
        valores[i] = i % 1000;
        indices[i] = i * 3;
    }

    for (int i = 0; i < 20000; i++) {
        acc += valores[i % 1000];
        acc += indices[i];
    }

    printf("%lld\n", acc);
    free(valores);
    free(indices);
    return 0;
}

File: benchmarks/jasboot_vs_v8/collections_large.js
let acc = 0;
const valores = [];
const indices = new Map();

for (let i = 0; i < 20000; i++) {
  valores.push(i % 1000);
  indices.set(i, i * 3);
}

for (let i = 0; i < 20000; i++) {
  acc += valores[i % 1000];
  acc += indices.get(i);
}

console.log(acc);
